### PR TITLE
feat: improved DX for unified-desktop-gui

### PR DESCRIPTION
**Pull request by tgriesser**
- Moves `graphql-codegen` config to the root, which will serve all packages needing it
- Adds gulpfile for coordinating scripts related to dev environment in launchpad app
- `yarn dev` from the root runs `yarn gulp dev`, which:
  - Runs [autobarrel](https://github.com/tgriesser/autobarrel) for rolling up the `@packages/graphql` files
  - Cleans the `dist` & cache for `.vite`
  - Starts the a codegen watcher for Nexus
  - Starts the `graphql-codegen --watch` & highlights output
  - Starts `vite` servers for `launchpad` & `app`
  - Starts electron `watch.js`
  
All of these streams are prefixed with the source so it's easy to know what's generating what:

image

  
Ignore any commented out/unused code, it's likely something that I either have plans to use or will come back to cleanup later.

TODO (separate PRs):
- Add dedicated `yarn debug` which will set proper debug ports across all processes
- Growl type notifications for errors

### CODE REVIEW

File: packages/graphql/src/entities/index.ts [hunk 1]
```diff
@@ -1,27 +1,20 @@
...
 export * from './Viewer'
-
 export * from './Wizard'
-
 export * from './WizardBundler'
-
 export * from './WizardFrontendFramework'
-
 export * from './WizardNpmPackage'
+export * from './run/'
```
Comment (JessicaSachs) [inline packages/graphql/src/entities/index.ts:20]: Is this a bug in autobarrel?
```suggestion
export * from './run'
```

File: packages/graphql/src/schema.ts [hunk 2]
```diff
@@ -16,12 +16,12 @@ const dirname = typeof __dirname !== 'undefined' ? __dirname : ''
 // for vite
 process.cwd ??= () => ''
 
-const isCodegen = Boolean(process.env.GRAPHQL_CODEGEN)
+const isCodegen = Boolean(process.env.CYPRESS_NEXUS_CODEGEN)
```
Comment (JessicaSachs) [inline packages/graphql/src/schema.ts:19]: `CYPRESS_INTERNAL` for Cy build + dev utils. `CYPRESS_` for user-modifiable environment variables.
```suggestion
const isCodegen = Boolean(process.env.CYPRESS_INTERNAL_NEXUS_CODEGEN)
```

File: scripts/gulp/gulpConstants.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import getenv from 'getenv'
+
+// Where to fetch the remote "federated" schema TODO: add w/ stitching PR
+// export const CYPRESS_CLOUD_ENV = 'production'
+
+export const CYPRESS_DEBUG_PORT_STARTUP = getenv.int('CYPRESS_DEBUG_PORT_STARTUP', 7200)
+
+export const CYPRESS_DEBUG_PORT_ELECTRON = getenv.int('CYPRESS_DEBUG_PORT_ELECTRON', 7201)
+
+export const CYPRESS_DEBUG_PORT_CODEGEN = getenv.int('CYPRESS_DEBUG_PORT_CODEGEN', 7202)
```
Comment (JessicaSachs) [inline scripts/gulp/gulpConstants.ts:10]: Ditto. `CYPRESS_INTERNAL_*`

File: scripts/gulp/gulpfile.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    // Now that we have the codegen, we can start the frontend(s)
+    gulp.parallel(
+      viteApp,
+      viteLaunchpad,
+    ),
+
+    // And we're finally ready for electron, watching for changes in /graphql to auto-restart the server
+    startElectronWatch,
+  ),
+)
```
Comment (JessicaSachs) [inline scripts/gulp/gulpfile.ts:38]: So, so beautiful. 😻

File: scripts/gulp/tasks/gulpTsc.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
+/* eslint-disable no-console */
```
Comment (JessicaSachs) [inline scripts/gulp/tasks/gulpTsc.ts:1]: Not that it matters too much, but, maybe a `.eslintrc` for the build directory?

File: scripts/gulp/utils/childProcessUtils.ts [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    await new Promise((ready) => setTimeout(ready, 100))
+  }
+
+  return true
+}
+
+export type AllSpawnableApps =
+  | `vite-${string}`
+  | `vite:build-${string}`
+  | 'gql-codegen'
```
Comment (JessicaSachs) [inline scripts/gulp/utils/childProcessUtils.ts:31]: Is this valid code? Github isn't syntax highlighting it because of the template strings.

**Review by JessicaSachs — CHANGES_REQUESTED** [review]
In general, running Cypress in global mode from the root of the Cypress repo (via `yarn dev`) looks good. I have a nit or two and a q about autobarrel maybe having a bug.

However, running w a specific projectRoot isn't working. The projectRoot is resolving relative to the server directory.

Here's the command I'm running from the root of the Cypress repo.
```sh
# I expect this to work.
yarn dev --project npm/vue
```

Once that's fixed and the right CYPRESS_INTERNAL env var prefixes are used, this is good to go.

File: packages/graphql/src/entities/index.ts (hunk 1 above)
Comment (tgriesser) [reply packages/graphql/src/entities/index.ts:20]: No, this is intentional. Trailing slash visually indicates that it's a directory

File: scripts/gulp/utils/childProcessUtils.ts (hunk 6 above)
Comment (tgriesser) [reply scripts/gulp/utils/childProcessUtils.ts:31]: Yep. This is https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html - newer feature, GH is probably lagging 😄

File: scripts/gulp/tasks/gulpTsc.ts (hunk 5 above)
Comment (tgriesser) [reply scripts/gulp/tasks/gulpTsc.ts:1]: Yeah, was thinking about this but got lazy using the autosuggest. Will add

**Pull request merged**